server/src/internalClusterTest/java/org/elasticsearch/gateway/GatewayIndexStateIT.java

```diff
@@ -523,28 +526,44 @@ public void testHalfDeletedIndexImport() throws Exception {
 
         final Metadata metadata = internalCluster().getInstance(ClusterService.class).state().metadata();
         final Path[] paths = internalCluster().getInstance(NodeEnvironment.class).nodeDataPaths();
-//        writeBrokenMeta(metaStateService -> {
-//            metaStateService.writeGlobalState("test", Metadata.builder(metadata)
-//                // we remove the manifest file, resetting the term and making this look like an upgrade from 6.x, so must also reset the
-//                // term in the coordination metadata
-//                .coordinationMetadata(CoordinationMetadata.builder(metadata.coordinationMetadata()).term(0L).build())
-//                // add a tombstone but do not delete the index metadata from disk
-//               .putCustom(IndexGraveyard.TYPE, IndexGraveyard.builder().addTombstone(metadata.index("test").getIndex()).build()).build());
-//            for (final Path path : paths) {
-//                try (Stream<Path> stateFiles = Files.list(path.resolve(MetadataStateFormat.STATE_DIR_NAME))) {
-//                    for (final Path manifestPath : stateFiles
-//                        .filter(p -> p.getFileName().toString().startsWith(Manifest.FORMAT.getPrefix())).collect(Collectors.toList())) {
-//                        IOUtils.rm(manifestPath);
-//                    }
-//                }
-//            }
-//        });
+        final String nodeId = client().admin().cluster().prepareNodesInfo(nodeName).clear().get().getNodes().get(0).getNode().getId();
+
+        writeBrokenMeta(metaStateService -> {
+            for (final Path path : paths) {
+                try (Stream<Path> stateFiles = Files.list(path.resolve(MetadataStateFormat.STATE_DIR_NAME))) {
```

Comment:
Think you can just say `IOUtils.rm(path.resolve(PersistedClusterStateService.METADATA_DIRECTORY_NAME));` since the `MetaStateService#writeGlobalState()` will create the directory again. Also preferring `PersistedClusterStateService.METADATA_DIRECTORY_NAME` over `MetadataStateFormat.STATE_DIR_NAME` to make it clear that we're wiping the Lucene-based metadata.

Reply:
thanks, fixed in f0e72e20e673399807bda0525b0c75dd272c150c